src/graphics/webgl/webgl-graphics-device.js

```diff
@@ -823,7 +824,18 @@ class WebglGraphicsDevice extends GraphicsDevice {
         // iOS exposes this for half precision render targets on both Webgl1 and 2 from iOS v 14.5beta
         this.extColorBufferHalfFloat = getExtension("EXT_color_buffer_half_float");
 
+        const ext = this.extDebugRendererInfo;
+        this.unmaskedRenderer = ext ? gl.getParameter(ext.UNMASKED_RENDERER_WEBGL) : '';
+        this.unmaskedVendor = ext ? gl.getParameter(ext.UNMASKED_VENDOR_WEBGL) : '';
+
         this.supportsInstancing = !!this.extInstancing;
+
+        // Check if we support GPU particles. At the moment, Samsung devices with Exynos (ARM) either crash or render
+        // incorrectly when using GPU for particles. See:
+        // https://github.com/playcanvas/engine/issues/3967
+        // https://github.com/playcanvas/engine/issues/3415
+        const samsungModelRegex = /SM-[a-zA-Z0-9]+\)/;
+        this.supportsGpuParticles = !(this.unmaskedVendor === 'ARM' && userAgent.match(samsungModelRegex));
```

Comment:
Should this be moved to `initializeCapabilities` instead? I placed it here because I saw `this.supportsInstancing` being done here but looking at it again, it makes more sense to move to `initializeCapabilities` call instead?

Reply:
yes, I assumed that's where it should go too.

Reply:
Moved!